compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1644,6 +1664,7 @@ private ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLook
             case TYPE_CAST_EXPRESSION:
             case TABLE_CONSTRUCTOR:
             case KEY_SPECIFIER:
+            case ERROR_TYPE_DESCRIPTOR:
```

Comment:
IIUC, though error-type-desc context is started here, it is ended only in one-of the paths. i.e: context will not be ended in the `error` case (without type-param).

Reply:
Fixed, That case handled as a simple type. Check if that is ok.